webapp/store/views.py

```diff
@@ -131,27 +138,20 @@ def user_details(username):
 @jaasstore.route(
     "/u/<username>/<charm_or_bundle_name>/<series_or_version>/<version>"
 )
-def user_entity(
-    username, charm_or_bundle_name, series_or_version=None, version=None
-):
+def user_entity(username, charm_or_bundle_name):
     charmhub_url = (
         "https://charmhub.io/" + username + "-" + charm_or_bundle_name
     )
-    return details(
-        charm_or_bundle_name,
-        series_or_version=series_or_version,
-        version=version,
-        charmhub_url=charmhub_url,
-    )
+    return redirect(charmhub_url, code=301)
 
 
 @jaasstore.route("/<charm_or_bundle_name>")
 @jaasstore.route("/<charm_or_bundle_name>/<series_or_version>")
 @jaasstore.route("/<charm_or_bundle_name>/<series_or_version>/<version>")
 def details(
     charm_or_bundle_name,
-    series_or_version=None,
-    version=None,
+    series_or_version,
+    version,
     charmhub_url=None,
 ):
     reference = None
```

Comment:
I can't add a comment but you'll want to remove the lines `158-167` and let charmhub handle the 404s. Make the redirect as simple as possible.

```
   try:
        reference = references.Reference.from_jujucharms_url(request.path[1:])
    except ValueError:
        pass
    entity = None
    if reference:
        entity = models.get_charm_or_bundle(reference)
if entity:
```